zerver/tests/test_user_groups.py

```diff
@@ -2037,6 +2121,31 @@ def test_updating_subgroups_of_user_group(self) -> None:
         self.assert_json_error(result, 'Nothing to do. Specify at least one of "add" or "delete".')
         self.assert_subgroup_membership(support_group, [leadership_group])
 
+        # Do not have support group as subgroup of any group to follow
+        # the condition a group used as a subgroup cannot be deactivated.
+        params = {"delete": orjson.dumps([support_group.id]).decode()}
+        result = self.client_post(f"/json/user_groups/{test_group.id}/subgroups", info=params)
+        self.assert_json_success(result)
+        self.assert_subgroup_membership(test_group, [])
+
+        # Test adding or removing subgroups from a deactivated group.
+        do_deactivate_user_group(support_group, acting_user=None)
+
+        params = {"delete": orjson.dumps([leadership_group.id]).decode()}
+        result = self.client_post(f"/json/user_groups/{support_group.id}/subgroups", info=params)
+        self.assert_json_error(result, "User group is deactivated.")
+        self.assert_subgroup_membership(support_group, [leadership_group])
+
+        params = {"add": orjson.dumps([test_group.id]).decode()}
+        result = self.client_post(f"/json/user_groups/{support_group.id}/subgroups", info=params)
+        self.assert_json_error(result, "User group is deactivated.")
+        self.assert_subgroup_membership(support_group, [leadership_group])
+
+        # Test that a deactivated group cannot be used as a subgroup.
+        params = {"add": orjson.dumps([support_group.id]).decode()}
+        result = self.client_post(f"/json/user_groups/{test_group.id}/subgroups", info=params)
+        self.assert_json_error(result, "User group is deactivated.")
```

Comment:
Error message here and in the test for checking user group is used as a setting or not, can be confusing with respect to which group is deactivated. But since we use the `has_user_group_access` function, the error message is this. We can either just add a separate if-condition for deactivation case and have a clear error message explaining which group is deactivated.

Reply:
Probably it's easy to debug in practice. I think it's possible a `code` would be a better way to indicate this than yet another string, if we decide it's important, but this seems OK to procrastinate.